src/Workspaces/Core/Portable/Workspace/Solution/SolutionCompilationState_Checksum.cs

```diff
@@ -158,30 +156,27 @@ public async Task<Checksum> GetChecksumAsync(ProjectId projectId, CancellationTo
         {
             throw ExceptionUtilities.Unreachable();
         }
+    }
 
-        static Checksum GetVersionMapChecksum(SolutionCompilationState @this)
-        {
-            // We want the projects in sorted order so we can generate the checksum for the
-            // source-generation-execution-map consistently.
-            var sortedProjectIds = SolutionState.GetOrCreateSortedProjectIds(@this.SolutionState.ProjectIds);
-            var supportedCount = sortedProjectIds.Count(
-                static (projectId, @this) => RemoteSupportedLanguages.IsSupported(@this.SolutionState.GetRequiredProjectState(projectId).Language),
-                @this);
-
-            // For each project, we'll add one checksum for the project id and one for the version map.
-            using var _ = ArrayBuilder<Checksum>.GetInstance(2 * supportedCount, out var checksums);
+    public SourceGeneratorExecutionVersionMap GetFilteredSourceGenerationExecutionMap(ProjectCone? projectCone)
```

Comment:
>public SourceGeneratorExecutionVersionMap GetFilteredSourceGenerationExecutionMap(ProjectCone? projectCone) [](http://example.com/codeflow?start=4&length=107)

Was this part of the change a functional fix, or something else?

Reply:
this was very much part of the functional fix.  When we're producing the checksum on the host side, we want to filter the map to what's in the cone.  Then, on the oop side, we want to be able to ask for this filtered map portion.  We then update that part of the OOP side with this filtered set.